src/menu/ExportIEDParams.ts

```diff
@@ -0,0 +1,303 @@
+        if (bType === 'Struct') {
+          const type = lastElement.getAttribute('type') ?? '';
+          return this.doc.querySelector(`DAType[id="${type}"]`);
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Search for the DO/SDO/DA/BDA element in the Template section of the document using the path array passed.
+   * The LN element is the starting point for the search in the Template section.
+   *
+   * @param lnElement         - The LN Element used as starting point in the Template section.
+   * @param dataAttributePath - The list of elements to search for, the names of the elements.
+   */
+  private getDataAttributeTemplateValue(
+    lnElement: Element,
+    dataAttributePath: string[]
+  ): string | null {
+    // This is only useful if the element to start from is the LN(0) Element.
+    if (
+      ['LN', 'LN0'].includes(lnElement.tagName) &&
+      dataAttributePath.length >= 2
+    ) {
+      // Search LNodeType Element that is linked to the LN(0) Element.
+      const type = lnElement.getAttribute('lnType');
+      let typeElement = this.doc.querySelector(`LNodeType[id="${type}"]`);
+      let lastElement: Element | null = null;
+
+      // Now start search through the Template section jumping between the type elements.
+      dataAttributePath.forEach(name => {
+        if (typeElement) {
+          lastElement = this.getDataElement(typeElement, name);
+          typeElement = this.getTypeElement(lastElement);
+        }
+      });
+
+      if (lastElement) {
+        const valElement = (<Element>lastElement).querySelector('Val');
+        return valElement?.textContent ?? null;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Search for the DAI element below the LN element using the path passed. The list of names is converted
+   * to a CSS Selector to search for the DAI Element and its Val Element.
+   *
+   * @param lnElement         - The LN Element used as starting point for the search.
+   * @param dataAttributePath - The names of the DOI/SDI/DAI Elements to search for.
+   */
+  private getDataAttributeInstanceValue(
+    lnElement: Element,
+    dataAttributePath: string[]
+  ): string | null {
+    if (
+      ['LN', 'LN0'].includes(lnElement.tagName) &&
+      dataAttributePath.length >= 2
+    ) {
+      const daiSelector = dataAttributePath
+        .map((path, index) => {
+          if (index === 0) {
+            // The first element is always a DOI element.
+            return `DOI[name="${path}"]`;
+          } else if (index === dataAttributePath.length - 1) {
+            // The last element is always a DAI element.
+            return `DAI[name="${path}"]`;
+          }
+          // Every element(s) between the DOI and DAI element is always a SDI element.
+          return `SDI[name="${path}"]`;
+        })
+        .join(' > ');
+
+      const daiValueElement = lnElement.querySelector(daiSelector + ' Val');
+      return daiValueElement?.textContent ?? null;
```

Comment:
For read-only not really a big issue. But do you think it would make sense to trim here?

Reply:
Done